xbmc/threads/platform/pthreads/ThreadImpl.cpp

```diff
@@ -72,18 +72,15 @@ void CThread::SetThreadInfo()
   m_ThreadOpaque.LwpId = syscall(SYS_gettid);
 #endif
 
-#if defined(HAVE_PTHREAD_SETNAME_NP)
-#ifdef TARGET_DARWIN
-#if(__MAC_OS_X_VERSION_MIN_REQUIRED >= 1060 || __IPHONE_OS_VERSION_MIN_REQUIRED >= 30200)
-  pthread_setname_np(m_ThreadName.c_str());
-#endif
-#else
-  pthread_setname_np(m_ThreadId, m_ThreadName.c_str());
+#if HAVE_PTHREAD_SETNAME_NP
+  // Add prefix so Kodi-owned threads are distinguishable from other libraries
+  std::string pthreadName("k/");
+  pthreadName += m_ThreadName;
+  // Limit on Linux is 15 chars + NULL
+  pthreadName.resize(15);
```

Comment:
darwin pthread names have no such limits

Reply:
This PR is about Linux. If you want to fix it for Darwin, be my guest :-)

Reply:
Actually it looks like you deleted the gating for Darwin and IOS which looks like it has a different method signature. Since this built without error I'm probably reading it wrong but if this is only intended for Linux I'm not sure why the Darwin/IOS section was deleted.

Reply:
There is nothing that sets the `#define` on Darwin since the switch to CMake, so the code is never used. I cannot fix it since I cannot build on Darwin. As I said, if someone wants to fix it for other OSes, go ahead. At the moment it's just dead code.

Reply:
Ripping out functionality for non-targeted platforms in PRs seems strange but if that is the way development is done now, your ball. I expect eventually, Darwin will become so gutted that it will get dropped.

Reply:
TARGET_DARWIN is defined

https://github.com/xbmc/xbmc/blob/master/cmake/scripts/osx/ArchSetup.cmake#L9
https://github.com/xbmc/xbmc/blob/master/CMakeLists.txt#L260

Reply:
@pkerling if you fix it so Darwin and IOS work without the limit like it used to, I'll merge it.

Reply:
> Ripping out functionality for non-targeted platforms in PRs seems strange but if that is the way development is done now, your ball.

Please stay on-topic. I am quite sure it *does not work* at the moment (see below). As such, I am not removing any currently existing/working functionality.

> TARGET_DARWIN is defined

This is not about `TARGET_DARWIN` at all. It is about `HAVE_PTHREAD_SETNAME_NP`. Where is that defined? Maybe this is magically defined somehow on Darwin and I just don't know - then by all means please do tell.

Reply:
autotools defined HAVE_PTHREAD_SETNAME_NP for DARWIN as it defined it for Linux. If you want to fix the issue only for Linux, fine, but not by removing code for other platforms. Code only inactive due to a bug should not be removed.

Reply:
> If you want to fix the issue only for Linux, fine, but not by removing code for other platforms. Code only inactive due to a bug should not be removed.

I was hoping that someone that actually has OS X would step up and fix it. It's really not much but without having OS X there is not much sense in trying. Seems @Razzeee will be kind enough to do that, so let's wait for a bit.

Reply:
Oops I meant @Rechi but anyway :-D

Reply:
I have OSX too and can help with non-cmake related changes. cmake only if you tell me what to do

Reply:
actually, this should probably be moved to common/ArchSetup.cmake. It's a test after all and will just fail on platforms that don't support it.

Reply:
I am against moving this to common. That any platform has `pthread_setname_np` does *not* mean that it works the way it works on Linux, or even takes the same arguments in the same order. Linux (glibc, rather, but I guess it's the only libc we support/test on) needs `_GNU_SOURCE` defined. Darwin very probably does not. BSD has `pthread_set_name_np`. This is highly platform-specific and so IMO is fine in `<platform>/ArchSetup.cmake`. For Darwin it's just 3 lines  or so anyway.

Reply:
Different suggestion: I think we could skip the CMake check altogether and do `#if TARGET_LINUX` or `#if TARGET_DARWIN` instead if we just always define `_GNU_SOURCE` on Linux. Do we support anything but glibc?

Reply:
we do support other libcs, so we need the check

Reply:
OK, then check stays in. I'd still like to not have it in `common` though.

Reply:
ok for me. We have worse code duplication in other parts ;)